### ISSUE TITLE
IE 10 DI doesn't work in es5 JIT mode with 9.0.0-rc.4/rc.5

### ISSUE THREAD

**Issue by maxokorokov** [angular/angular#34249]
# 🐞 bug report

### Affected Package

Looks like `@angular/core` related to injection in IE 10 in JIT mode.

Also looks like resurfaced https://github.com/angular/angular-cli/issues/14888

### Is this a regression?

Yes, latest Angular 8 works fine.

### Description

Simplest DI use case fails in IE 10 with es5 target in testing or serving with JIT
IE 11 works fine.

## 🔬 Minimal Reproduction

Steps:
- create a new cli project with `9.0.0-rc.4 or 5`
- add any service and inject it in the main component
- change `target` to `es5`
- run `ng test` or `ng serve` (in JIT mode with `aot: 'false'`)

Example:

```ts
import {Component, Injectable} from '@angular/core';

@Injectable()
export class Service {}

@Component({
  selector: 'app-root',
  template: `<h1>Hello world</h1>`,
  providers: [Service]
})
export class AppComponent {
  constructor(service: Service) {}
}
```

Repo to clone with the steps above → https://github.com/maxokorokov/ie-injection-fail

## 🔥 Exception or Error
<pre><code>
TypeError: Unable to get property 'ngMetadataName' of undefined or null reference
	error properties: Object({ number: -2146823281 })
	    at <Jasmine>
	   at reflectDependency (http://172.16.220.123:9876/_karma_webpack_/vendor.js:48262:18)
	   at Anonymous function (http://172.16.220.123:9876/_karma_webpack_/vendor.js:48237:38)
	   at Anonymous function (http://172.16.220.123:9876/_karma_webpack_/polyfills-es5.js:888:7)
	   at map (http://172.16.220.123:9876/_karma_webpack_/polyfills-es5.js:5298:5)
	   at convertDependencies (http://172.16.220.123:9876/_karma_webpack_/vendor.js:48237:5)
	   at reflectDependencies (http://172.16.220.123:9876/_karma_webpack_/vendor.js:48233:5)
	   at directiveMetadata (http://172.16.220.123:9876/_karma_webpack_/vendor.js:62546:5)
	   at get (http://172.16.220.123:9876/_karma_webpack_/vendor.js:62455:17)
	   at getComponentDef (http://172.16.220.123:9876/_karma_webpack_/vendor.js:37811:5)
	   at verifyDeclarationsHaveDefinitions (http://172.16.220.123:9876/_karma_webpack_/vendor.js:62135:9)
</code></pre>

Debugging fails here:

![Screen Shot 2019-12-05 at 11 39 15](https://user-images.githubusercontent.com/8074436/70228071-03671380-1754-11ea-9ae7-78f07a6f031c.png)

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 9.0.0-rc.5
Node: 12.12.0
OS: darwin x64

Angular: 9.0.0-rc.5
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-rc.5
@angular-devkit/build-angular     0.900.0-rc.5
@angular-devkit/build-optimizer   0.900.0-rc.5
@angular-devkit/build-webpack     0.900.0-rc.5
@angular-devkit/core              9.0.0-rc.5
@angular-devkit/schematics        9.0.0-rc.5
@ngtools/webpack                  9.0.0-rc.5
@schematics/angular               9.0.0-rc.5
@schematics/update                0.900.0-rc.5
rxjs                              6.5.3
typescript                        3.6.4
webpack                           4.41.2
</code></pre>

**Comment by gkalpak** [comment]
I am probably stating the obvious, but the error is caused by JIT's `reflectDependency()` util trying to access `__proto__`, which [accoding to MDN][1] is not supported in IE<11:

https://github.com/angular/angular/blob/0aff4a6919e48dd6425fe6e53d5b867c9a09a791/packages/core/src/di/jit/util.ts#L54-L60

<sub>So, the question is why does it do this naughty thing (accessing `__proto__` is considered a bad practice) instead of using [Object.getPrototypeOf()][2] (which is supported in IE10+ and polyfilled by `core-js` afaict) or the `in` operator, so something.</sub>

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/proto
[2]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getPrototypeOf

**Comment by pkozlowski-opensource** [comment]
Note to self: verify why this wasn't caught by any of our test suites.

**Comment by AndriiDidkivsky** [comment]
Please, stop to support this ancient monster. 
[Support ending for Internet Explorer 10](https://support.microsoft.com/en-us/help/4488955/support-ending-for-internet-explorer-10)

**Comment by pkozlowski-opensource** [comment]
This issue was fixed in master (#34305), will be part of the next RC release.
